src/pages/index.js

```diff
@@ -101,20 +99,72 @@ function operatePopup(popup, action="remove") {
   }
 }
 
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId; 
+  operatePopup(deleteModal, "add");
+}
+
+function handleDeletion(evt) {
+  evt.preventDefault();
+
+  const deltetionButton = evt.submitter;
+  deltetionButton.textContent = "Deleting...";
+
+  api.deleteCard(selectedCardId)
+  .then(() => {
+    selectedCard.remove();
+    operatePopup(deleteModal);
+  })
+  .catch((error) => {
+    console.error(error);
+  })
+  .finally(() => {
+    deltetionButton.textContent = "Delete";
+  });
+}
+
+function toggleLikeStatus(cardId, isLiked) {
+  if (isLiked) {
+    api.dislikeCard(cardId)
+    .then(() => {
+      
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+  }
+  else {
+    api.likeCard(cardId)
+    .then(() => {
+      
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+  }
+}
+
 // Function to create a new card element
-function createCard(item) {
+function createCard(data) {
   const cardElement = cardTemplate.cloneNode(true).firstElementChild;
   const cardImage = cardElement.querySelector(".card__content-image");
   const cardCaption = cardElement.querySelector(".card__content-caption");
   const cardLikeButton = cardElement.querySelector(".card__content-like");
   const cardDeleteButton = cardElement.querySelector(".card__delete-button");
 
-  cardImage.src = item.link;
-  cardImage.alt = item.name;
-  cardCaption.textContent = item.name;
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+  cardCaption.textContent = data.name;
 
   // Toggle like button state
   cardLikeButton.addEventListener("click", () => {
+    const isLiked = cardLikeButton.classList.contains("card__content-like_liked");
+
+    console.log(`isLiked: ${isLiked}`);
+
+    toggleLikeStatus(data._id, isLiked);
+
     cardLikeButton.classList.toggle("card__content-like_liked");
```

Comment:
You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.